src/humanloop/client.py

```diff
@@ -136,22 +135,20 @@ def __init__(
             )
         instrument_provider(provider=self._tracer_provider)
         self._tracer_provider.add_span_processor(
-            HumanloopSpanProcessor(
-                client=self,
-                exporter=HumanloopSpanExporter(client=self),
-            ),
+            HumanloopSpanProcessor(exporter=HumanloopSpanExporter(client=self)),
         )
 
         if opentelemetry_tracer is None:
             self._opentelemetry_tracer = self._tracer_provider.get_tracer("humanloop.sdk")
         else:
             self._opentelemetry_tracer = opentelemetry_tracer
 
+    @contextmanager
```

Comment:
as discussed earlier, this is cool but probably unnecessary -- simpler to just keep it as a decorator on a callable